e2e/__tests__/testEnvironmentCircusAsync.test.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {skipSuiteOnJasmine} from '@jest/test-utils';
+import runJest from '../runJest';
+
+skipSuiteOnJasmine();
+
+it('calls asynchronous handleTestEvent in testEnvironment', () => {
+  const result = runJest('test-environment-circus-async');
+  expect(result.failed).toEqual(true);
+
+  const lines = result.stdout.split('\n');
+  expect(lines).toMatchInlineSnapshot(`
+    Array [
+      "setup",
+      "warning: add_hook is a sync event",
```

Comment:
@SimenB 

> could we verify this in code?

Well, at the very least, I verify now this in E2E tests.

Reply:
wonderful, thanks!